react/index.js

```diff
@@ -74,34 +75,43 @@ class MiniCart extends Component {
     return partition(isNotInCart, clientItems)
   }
 
-  handleItemsDifference = async clientItems => {
-    const { showToast, intl } = this.props
-    this.setState({ updatingOrderForm: true })
+  handleItemsDifference = modifiedItems => {
+    this.setState({ updatingOrderForm: true }, () => this.sendModifiedItemsToServer(modifiedItems))
+  }
+
+  sendModifiedItemsToServer = async modifiedItems => {
+    const { showToast, intl, updateItemsSentToServer } = this.props
+    const [itemsToAdd, itemsToUpdate] = this.partitionItemsAddUpdate(modifiedItems)
+    await updateItemsSentToServer()
+    const pickProps = map(pick(['id', 'index', 'quantity', 'seller', 'options']))
     try {
-      const [itemsToAdd, itemsToUpdate] = this.partitionItemsAddUpdate(clientItems)
-      const pickProps = map(pick(['id', 'index', 'quantity', 'seller', 'options']))
       const updateItemsResponse = await this.updateItems(pickProps(itemsToUpdate))
       const addItemsResponse = await this.addItems(pickProps(itemsToAdd))
+      itemsToAdd.length > 0 && this.props.push({
+          event: 'addToCart',
+          items: itemsToAdd,
+        })
+      const modifiedItems = this.getModifiedItemsOnly()
```

Comment:
doesnt this var name conflict with the one that comes from the function argument?